src/manifests/input_manifest.py

```diff
@@ -130,6 +130,13 @@ def stable(self) -> 'InputManifest':
         manifest.components.__stabilize__()
         return manifest
 
+    def plugins_depend_on(self, plugin: str) -> List[str]:
+        plugins = []
+        for component in self.components.select():
+            if component.depends_on is not None and plugin in component.depends_on:
```

Comment:
This assumes that each component in the input_manifest has `depends_on` key. Is the the case?
If not, it should ideally check if the component has `depends_on` key and then process further. 
Also, please add test for this.

Reply:
A little confused. It firstly check `depends_on` key is present and then check if the current component is depending on the given plugin.

Reply:
Okay, I thought `component.depends_on is not None` is checking if the value of `depends_on` key. 
Please add the test.

Reply:
Test is added in the latest commit.